streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -115,7 +115,7 @@
      * </pre>
      */
     public enum State {
-        CREATED(1), RUNNING(1, 2, 4), PARTITIONS_REVOKED(3, 4), ASSIGNING_PARTITIONS(1, 4), PENDING_SHUTDOWN(5), DEAD;
+        CREATED(1, 4), RUNNING(1, 2, 4), PARTITIONS_REVOKED(3, 4), ASSIGNING_PARTITIONS(1, 4), PENDING_SHUTDOWN(5), DEAD;
```

Comment:
Just wondering, if there could be an error that forces us to go from `ASSIGNING_PARTITIONS` to `PENDING_SHUTDOWN` directly ?